### ISSUE TITLE
[Question] Learning with a quantized model

### ISSUE THREAD

**Issue by chunky** [DLR-RM/stable-baselines3#667]
### Question

I'm doing training using a custom gym and SB3 and getting really great results. Part of my endgame is to run the final inference on a Coral for the forward pass, so it can be run on a raspberry pi.

The coral documentation, here, talks about quantized models: https://coral.ai/docs/edgetpu/models-intro/#quantization

In particular, it talks about quantization-aware training:

> For compatibility with the Edge TPU, you must use either quantization-aware training (recommended) or full integer post-training quantization. [...] Thus, this technique requires modification to the network before initial training. This generally results in a higher accuracy model (compared to post-training quantization)

So, the question is, is there a good path to doing this with SB3? The net_arch param to kwargs doesn't really afford that level of nuance, that I am able to discern?

### Additional context

I'm slowly working through the alternative process, of exporting to ONNX, converting to TFLite, then converting to Coral, but I'm slightly out of my depth.

### Checklist

- [*] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [*] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)

**Comment by Miffyli** [comment]
Hey. There is no pre-made option for this. Sounds like one option could be to create a custom policy (see docs) which defines a network with this quantization enabled.

Alternatively, if this is possible, you could recreate the network in you target platform with the necessary libraries, and then load of the parameters from the SB3 agent manually by first exporting them to torch object  / numpy arrays (`get_parameters` function) and then importing them on your target platform. This would avoid the hassle with multiple libraries :)

**Comment by araffin** [comment]
> Part of my endgame is to run the final inference on a Coral for the forward pass, so it can be run on a raspberry pi.

Depending on the size of the policy network, you may not need Coral, and just use something simpler (PyTorch JIT, C++ runtime or ONNX) directly on the raspberry pi.

**Comment by chunky** [comment]
@araffin Currently I have the forward pass running on the pi, using torch and SB3 directly. The performance is great with small versions of my network, tolerable with larger versions, but I have grander aspirations, and the Coral seems like a really great solution.

@Miffyli I was looking through the docs, I see how to create a custom policy, now just need to figure out if Torch can actually help me with that. I looked through the Coral API; it doesn't seem to have an API to create networks directly, only to reinterpret tflite networks [which sort of makes sense, since the big G makes both Coral and TF]. So perhaps a direct get_parameters->tflite approach could work, too.

Thank-you both!

**Comment by araffin** [comment]
closed by https://github.com/DLR-RM/stable-baselines3/pull/679
